app/src/display/widgets/battery_status.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <bluetooth/services/bas.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display/widgets/battery_status.h>
+#include <zmk/usb.h>
+#include <zmk/events/usb-conn-state-changed.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/battery-state-changed.h>
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+static lv_style_t label_style;
+
+void battery_status_init() {
+    if (label_style.text.font != NULL) {
+        return;
+    }
+
+    lv_style_copy(&label_style, &lv_style_plain);
+    label_style.text.color = LV_COLOR_BLACK;
+    label_style.text.font = &lv_font_roboto_16;
+    label_style.text.letter_space = 1;
+    label_style.text.line_space = 1;
+}
+
+void set_battery_symbol(lv_obj_t *label) {
+    char text[2] = "  ";
+    u8_t level = bt_gatt_bas_get_battery_level();
+
+    if (zmk_usb_is_powered()) {
+        strcpy(text, LV_SYMBOL_CHARGE);
+    }
+
+    if (level > 95) {
+        strcat(text, LV_SYMBOL_BATTERY_FULL);
+    } else if (level > 65) {
+        strcat(text, LV_SYMBOL_BATTERY_3);
+    } else if (level > 35) {
+        strcat(text, LV_SYMBOL_BATTERY_2);
+    } else if (level > 5) {
+        strcat(text, LV_SYMBOL_BATTERY_1);
+    } else {
+        strcat(text, LV_SYMBOL_BATTERY_EMPTY);
+    }
```

Comment:
I like this a lot as the battery percent isn't super accurate. Will hopefully reduce people being confused by the percent going up and down somewhat randomly.

Reply:
Yeah, there's a reason I didn't just print the battery percentage as a number for the first pass on this.